cdap-docs/examples-manual/source/examples/log-analysis.rst

```diff
@@ -0,0 +1,189 @@
+    :description: Cask Data Application Platform SparkPageRank Application
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _examples-spark-page-rank:
+
+=======================
+Log Analysis Example
+=======================
+
+A Cask Data Application Platform (CDAP) example demonstrating Spark and MapReduce running in parallel inside a
+Workflow through fork.
+
+Overview
+========
+
+This example demonstrates Spark and MapReduce performing log analysis, computing total number of hits for every unique url, total number of responses for every unique response code and total number of requests made by every unique ip address based on Apache usage log.
+
+Logs are sent to the CDAP and ingested into the *logStream*, which stores the log information event in its entirety.
+
+After these events are streamed, they are taken up by the *ResponseCounterSpark*, which
+goes through the entries, calculates the total number of responses for every unique response code and tabulates results in an KeyValueTable dataset, *responseCount*.
+The Spark program also computes the total number of requests made by every unique ip address and writes it to TimePartitionedFileSet, *reqCount*.
+
+In parallel, these events are also taken up by the *HitCounterProgram*, which goes through the entries, calculates the total number hits for every unique url and tabulates results in KeyValueTable dataset, *hitCount*.
+
+The *LogAnalysisWorkflow* ties the Spark and MapReduce to run in parallel.
+
+Once the application completes, you can query the *responseCount* dataset by using the ``rescount`` endpoint of the *ResponseCounterService*.
+It will send back a string result with total number of responses on the ``rescount`` query parameter. You can also query the
+*hitCount* dataset by using ``url`` endpoint of the *HitCounterService*. It will send the total number of hits for the queried url.
+You can query the ``reqCount`` TimePartitionedFileSet by using the ``reqcount`` endpoint of the *RequestCounterService* which will return a set of all the available partitions.
+Using one of partitions from the above set you can query for the total number of request made by every unique ip address in last 60 minutes by using the ``reqfile``
+endpoint of the *RequestCounterService* which will return a Map of ip addresses and total number of requests made by them.
+
+Let's look at some of these components, and then run the application and see the results.
+
+The LogAnalysis Application
+-----------------------------
+
+As in the other `examples <index.html>`__, the components
+of the application are tied together by the class ``LogAnalysisApp``:
+
+.. literalinclude:: /../../../cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java
+   :language: java
+   :lines: 70-94
+
+The ``hitCount``, ``responseCount`` KeyValueTable and  ``reqCount`` TimePartitionedFileSet
+--------------------------------------------------------------------------------
```

Comment:
All underlines must be *at least* as long as the title above them. These two fall short. I would also suggest that the first title be:
```
The *hitCount* and  *responseCount* ``KeyValueTable``s, and the  *reqCount* ``TimePartitionedFileSet``
```
which should look something like:
The *hitCount* and  *responseCount* ``KeyValueTable``s, and the  *reqCount* ``TimePartitionedFileSet``

The idea is that class names are in ``literal font``, while named objects (such as services, streams, etc.) are in *italic font*.